cmd/celestia-appd/cmd/start.go

```diff
@@ -0,0 +1,632 @@
+			_ = app.Close()
+		}
+
+		if apiSrv != nil {
+			_ = apiSrv.Close()
+		}
+
+		ctx.Logger.Info("exiting...")
+	}()
+
+	// wait for signal capture and gracefully return
+	return server.WaitForQuitSignals()
+}
+
+func startTelemetry(cfg serverconfig.Config) (*telemetry.Metrics, error) {
+	if !cfg.Telemetry.Enabled {
+		return nil, nil
+	}
+	return telemetry.New(cfg.Telemetry)
+}
+
+// wrapCPUProfile runs callback in a goroutine, then wait for quit signals.
+func wrapCPUProfile(ctx *server.Context, callback func() error) error {
+	if cpuProfile := ctx.Viper.GetString(flagCPUProfile); cpuProfile != "" {
+		f, err := os.Create(cpuProfile)
+		if err != nil {
+			return err
+		}
+
+		ctx.Logger.Info("starting CPU profiler", "profile", cpuProfile)
+		if err := pprof.StartCPUProfile(f); err != nil {
+			return err
+		}
+
+		defer func() {
+			ctx.Logger.Info("stopping CPU profiler", "profile", cpuProfile)
+			pprof.StopCPUProfile()
+			if err := f.Close(); err != nil {
+				ctx.Logger.Info("failed to close cpu-profile file", "profile", cpuProfile, "err", err.Error())
+			}
+		}()
+	}
+
+	errCh := make(chan error)
+	go func() {
+		errCh <- callback()
+	}()
+
+	select {
+	case err := <-errCh:
+		return err
+
+	case <-time.After(srvrtypes.ServerStartTime):
+	}
+
+	return server.WaitForQuitSignals()
+}
+
+func addCommands(
+	rootCmd *cobra.Command,
+	defaultNodeHome string,
+	appCreator srvrtypes.AppCreator,
+	appExport srvrtypes.AppExporter,
+	addStartFlags srvrtypes.ModuleInitFlags,
+) {
+	tendermintCmd := &cobra.Command{
+		Use:     "tendermint",
+		Aliases: []string{"comet", "cometbft"},
+		Short:   "Tendermint subcommands",
+	}
+
+	tendermintCmd.AddCommand(
+		server.ShowNodeIDCmd(),
+		server.ShowValidatorCmd(),
+		server.ShowAddressCmd(),
+		server.VersionCmd(),
+		cmtcmd.ResetAllCmd,
+		cmtcmd.ResetStateCmd,
+		server.BootstrapStateCmd(appCreator),
+	)
+
+	startCmd := startCmd(appCreator, defaultNodeHome)
+	addStartFlags(startCmd)
+
+	rootCmd.AddCommand(
+		startCmd,
+		tendermintCmd,
+		server.ExportCmd(appExport, defaultNodeHome),
+		version.NewVersionCommand(),
+		server.NewRollbackCmd(appCreator, defaultNodeHome),
+	)
+}
+
+// checkBBR checks is bbr is configured to be used as a congestion control algo.
+func checkBBR(command *cobra.Command) error {
+	const (
+		errorMsg = `
+// The BBR congestion control algorithm does not appear to be enabled in this 
+// system's kernel. This is important for the p2p stack to be performant.
+//
+// to enable bbr call:
+//
+sudo modprobe tcp_bbr
+net.core.default_qdisc=fq
+net.ipv4.tcp_congestion_control=bbr
+sudo sysctl -p
+// 
+// and can be verified to be running using
+//
+sysctl net.ipv4.tcp_congestion_control
+
+// This might not work for all systems, you might have to search online to 
+// figure out how to enable bbr for your system.
+//
+// While this node will get worse performance using something other than bbr,
+// If you need to bypass this block use the "--force-no-bbr true" flag.
```

Comment:
depending on the parser, `--x true` can give different results, i.e. it will probably be parsed as `--x` and `true` as a positional argument

if you want to keep explicit value for it, then it should be `--force-no-bbr=true`, if it's supported by the args parser